DependencyInjection/Compiler/AddVotersPass.php

```diff
@@ -22,14 +22,28 @@ public function process(ContainerBuilder $container)
         $definition = $container->getDefinition('knp_menu.matcher');
         $listener = $container->getDefinition('knp_menu.listener.voters');
 
-        foreach ($container->findTaggedServiceIds('knp_menu.voter') as $id => $tags) {
-            $definition->addMethodCall('addVoter', array(new Reference($id)));
+        $voters = array();
 
+        foreach ($container->findTaggedServiceIds('knp_menu.voter') as $id => $tags) {
             foreach ($tags as $tag) {
+                $priority = isset($tag['priority']) ? (int) $tag['priority'] : 0;
+                $voters[$priority][] = $id;
+
                 if (isset($tag['request']) && $tag['request']) {
                     $listener->addMethodCall('addVoter', array(new Reference($id)));
                 }
             }
         }
+
+        if (empty($voters)) {
+            return;
+        }
+
+        krsort($voters);
+        $sortedVoters = call_user_func_array('array_merge', $voters);
+
+        foreach ($sortedVoters as $id) {
+            $definition->addMethodCall('addVoter', array(new Reference($id)));
```

Comment:
well, this is what I rejected in #268. I don't want to add the concept of priorities in the library itself.

If you want to add voters in your own compiler pass by modifying the definition directly, you can still manipulate the whole list of method calls of the Definition if you don't want to add it at the end of the list (same than what you would do today).

Do you really need to register a voter at runtime ? I never faced such scenario.